core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -158,15 +160,28 @@ public void registerQuery(
             Set<DynamicFilterId> lazyDynamicFilters,
             Set<DynamicFilterId> replicatedDynamicFilters)
     {
-        Map<DynamicFilterId, SettableFuture<Void>> lazyDynamicFilterFutures = lazyDynamicFilters.stream()
```

Comment:
Are we sure first `attemptId` is always 0?

Reply:
currentAttempt in SqlQueryScheduler starts with 0
@arhimondr please confirm if it's ok to assume this

Reply:
Yes, it starts with 0